parser/parser.y

```diff
@@ -11961,6 +11965,10 @@ StringType:
 	{
 		tp := types.NewFieldType(mysql.TypeMediumBlob)
 		tp.SetCharset($3.(*ast.OptBinary).Charset)
+		if $3.(*ast.OptBinary).Charset == charset.CharsetBin {
+			tp.AddFlag(mysql.BinaryFlag)
+			tp.SetCollate(charset.CollationBin)
+		}
```

Comment:
Instead of modify `OptCharsetWithOptBinary` and affect all terms depending on the struct, a special treament for  `NewFieldType(mysql.TypeMediumBlob)` makes more sense to me. What do you think?

Reply:
But it's much close to MySQL's parser.
https://github.com/mysql/mysql-server/blob/8d8c986e5716e38cb776b627a8eee9e92241b4ce/sql/sql_yacc.yy#L7655-L7697